fs/options.go

```diff
@@ -147,6 +147,10 @@ func FixRangeOption(options []OpenOption, size int64) {
 				x = &RangeOption{Start: size - x.End, End: -1}
 				options[i] = x
 			}
+			if x.End > size {
```

Comment:
Can you stick this in a separate commit please (doesn't need to be a new PR), update the comment for this function and justify why you need to make the change in the commit message - thanks!

Reply:
This one might actually need a bit of discussion. I added this as quick fix for passing the TestObjectOpenRange unit test. Specificly this one:
``{fs.RangeOption{Start: 81, End: 100000}, 81, 100}``
As far as I understand RFC7233 this request should actually fail with Range not Satisfiable and on Jottacloud it does. However all the other backends implemented so far seem to be fine with it.

Reply:
Interesting... It is a bit vague the RFC isn't it.  Especially this bit:

> Thus, clients cannot depend  on receiving a 416 (Range Not Satisfiable) response even when it       is most appropriate.

I'm happy for this to go in a separate commit and it will get run through all the integration tests for the other backends so we can see if it breaks anything else (I don't think it will).